pages/blog/typescript-types.mdx

```diff
@@ -0,0 +1,159 @@
+import withArticle from '../../ui/blog/article'
+export const meta = {
+  title: 'TypeScript basic types system',
+  author: 'tuvalSimha',
+  tags: ['graphql', 'typescript', 'codegen'],
+  date: '2022-07-16',
+  description: 'How to use Typescript basic types system',
+  image: '/blog-assets/graphql-code-generator/cover.png'
+}
+export default withArticle(meta)
+ 
+ 
+# Introduction
+
+TypeScript programming language is a superset of JavaScript that adds types to JavaScript using a set of tools called a type system.
```

Comment:
I think the introduction is too broad - we don't have to talk about TypeScript or what it is, we can assume people who landed on this page know what TypeScript is. 

I would prefer the introduction to cover what's union types and then move to the actual thing.